### PR TITLE
chore(auth): Added snippets for multitenancy and IdP management

### PR DESCRIPTION
**Pull request by hiranya911**
Snippets for:
* https://cloud.google.com/identity-platform/docs/managing-providers-programmatically
* https://cloud.google.com/identity-platform/docs/multi-tenancy-managing-tenants

### CODE REVIEW

File: src/test/java/com/google/firebase/snippets/FirebaseAuthSnippets.java [hunk 1]
```diff
@@ -657,14 +665,475 @@ public void generateSignInWithEmailLink() {
...
+    for (SamlProviderConfig config : page.iterateAll()) {
+      System.out.println(config.getProviderId());
+    }
+    // [END list_saml_providers]
+  }
+
+  public void createOidcProviderConfig() throws FirebaseAuthException {
+    // [START create_oidc_provider]
+    OidcProviderConfig.CreateRequest request = new OidcProviderConfig.CreateRequest()
+        .setDisplayName("SAML provider name")
```
Comment (lahirumaramba) [inline src/test/java/com/google/firebase/snippets/FirebaseAuthSnippets.java:733]:
```suggestion
        .setDisplayName("OIDC provider name")
```

File: src/test/java/com/google/firebase/snippets/FirebaseAuthSnippets.java [hunk 2]
```diff
@@ -657,14 +665,475 @@ public void generateSignInWithEmailLink() {
...
+    }
+    // [END import_without_password_tenant]
+  }
+
+  public void verifyIdTokenTenant(TenantAwareFirebaseAuth tenantAuth, String idToken) {
+    // [START verify_id_token_tenant]
+    try {
+      // idToken comes from the client app
+      FirebaseToken token = tenantAuth.verifyIdToken(idToken);
+      // This should be set to TENANT-ID. Otherwise TenantIdMismatch error thrown.
```
Comment (lahirumaramba) [inline src/test/java/com/google/firebase/snippets/FirebaseAuthSnippets.java:1038]: Should we clarify what we mean by `this` here?

File: src/test/java/com/google/firebase/snippets/FirebaseAuthSnippets.java (hunk 1 above)
Comment (hiranya911) [reply src/test/java/com/google/firebase/snippets/FirebaseAuthSnippets.java]: Good catch! Done.

File: src/test/java/com/google/firebase/snippets/FirebaseAuthSnippets.java (hunk 2 above)
Comment (hiranya911) [reply src/test/java/com/google/firebase/snippets/FirebaseAuthSnippets.java]: Done

**Pull request merged**